mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/KMEngine.m

```diff
@@ -311,7 +311,7 @@ - (NSArray *)processGroup:(KMCompGroup *)gp event:(NSEvent *)event {
             else {
                 //if (self.debugMode)
                 //NSLog(@"No shift flags!");
-                if (!key.context.length) {
+                if (!key.context.length && !(mask & K_MODIFIERFLAG)) {
```

Comment:
The additional test needs to be on line 311, because the else case on line 318 has exactly the same requirement.

Reply:
Agreed. Let me see if I can come up with a unit test to support the need for this...